### PR TITLE
[webgpu] fix compiling error

### PR DESCRIPTION
**Pull request by xhcao**

### CODE REVIEW

**Comment by xhcao** [comment]
@fs-eire @guschmue Please take a look, dawn removed all uses of `const char*` callbacks, https://dawn-review.googlesource.com/c/dawn/+/210437
@jchen10 @hujiajie

File: onnxruntime/core/providers/webgpu/buffer_manager.cc [hunk 1]
```diff
@@ -321,8 +321,8 @@ void BufferManager::Download(WGPUBuffer src, void* dst, size_t size) {
 
   // TODO: revise wait in whole project
 
-  ORT_ENFORCE(context_.Wait(staging_buffer.MapAsync(wgpu::MapMode::Read, 0, buffer_size, wgpu::CallbackMode::WaitAnyOnly, [](wgpu::MapAsyncStatus status, const char* message) {
-    ORT_ENFORCE(status == wgpu::MapAsyncStatus::Success, "Failed to download data from buffer: ", message);
+  ORT_ENFORCE(context_.Wait(staging_buffer.MapAsync(wgpu::MapMode::Read, 0, buffer_size, wgpu::CallbackMode::WaitAnyOnly, [](wgpu::MapAsyncStatus status, wgpu::StringView message) {
+    ORT_ENFORCE(status == wgpu::MapAsyncStatus::Success, "Failed to download data from buffer: ", message.data);
```
Comment (fs-eire) [inline onnxruntime/core/providers/webgpu/buffer_manager.cc:325]: `std::string_view{message}` should be better, as a string view may not necessarily be ended with '\0'

Comment (fs-eire) [reply onnxruntime/core/providers/webgpu/buffer_manager.cc:325]: see #23002

**Comment by fs-eire** [comment]
I am surprised that existing CI pipelines didn't catch the build error. @xhcao what is your build environment?

File: onnxruntime/core/providers/webgpu/buffer_manager.cc (hunk 1 above)
Comment (xhcao) [reply onnxruntime/core/providers/webgpu/buffer_manager.cc]: Done

**Comment by xhcao** [comment]
> I am surprised that existing CI pipelines didn't catch the build error. @xhcao what is your build environment?

I cannot produce this issue if I upgrade the Visual studio and the version VC is 14.42.34433.
I produced this issue on Debug build and old visual studio.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**